lib/hon_graffiti_phoenix/parsers/markup_parser.ex

```diff
@@ -0,0 +1,64 @@
+defmodule HonGraffitiPhoenix.Parsers.MarkupParser do
+  alias HonGraffitiPhoenix.Parsers.DecoratedString
+
+  @color_map %{
+    "w" => "white", "*" => "white", "r" => "red",
+    "b" => "blue", "y" => "yellow", "m" => "magenta",
+    "n" => "brown", "p" => "purple", "o" => "orange",
+    "t" => "teal", "v" => "grey", "g" => "green",
+    "k" => "black", "" => "black"}
```

Comment:
Re: this empty string key, I think if a caret is followed by an invalid color code this map should never be accessed and the caret should be rendered in the body (so this key/value pair should be unnecessary).

As a separate but related point, it looks like you have a test that verifies the inverse of that ("broken markdown won't be parsed into a color") but I think it might also be good to verify that the body includes the caret, as expected.